src/main/jbake/content/blog/2016/09/06/consuming-hawkular-api-over-ssl.adoc

```diff
@@ -0,0 +1,84 @@
+= Consuming Hawkular API over SSL with self signed certificates
+Josejulio Martínez
+2016-9-6
+:jbake-type: post
+:jbake-status: published
+:jbake-tags: blog
+
+In this post I'll show you how to trust self-signed certificates for use with Ruby client, used on projects such as ManageIQ and HawkFX.
+
+If you haven't already, follow this guide for
+link:http://www.hawkular.org/hawkular-services/docs/installation-guide/secure-comm.html[securing communications]
+ on your Hawkular server, because we'll make use of the files generated by that guide.
+
+As today, there are Hawkular clients for the following programming languages: Java, Python, Go and link:http://github.com/hawkular/hawkular-client-ruby[Ruby].
+The Hawkular Ruby client lets you connect using a secure connection. Buying a certificate from a trusted SSL vendor to host your
+own development server can be cost-prohibitive or just meaningless.
+
+If you are going to self-sign your own development certificates for use outside of java, you will need to do some additional steps
+to tell your consumer environment to trust them.
+
+For this you will need to:
+
+.  Install `c_rehash`.
+.  Identify where OpenSSL reads certificates and keys from.
+.  Hand over your certificate to OpenSSL in PEM format and rehash.
+
+== Install c_rehash
+`c_rehash` is an OpenSSL utility to create symbolic links to files by their hash values.
+This format is used by OpenSSL to find certificates.
+
+You can install it on fedora by executing
+[source, bash]
+----
+$ sudo dnf install openssl-perl
+----
+
+== Identify where OpenSSL reads certificates and keys from.
+OpenSSL reads certificates and keys from a specific path, you will need to copy your self-signed certificates there.
+You can find this path by running the following command:
+
+[source, bash]
+----
+$ openssl version -d
+OPENSSLDIR: "/etc/pki/tls"
+----
+
+In this case, OpenSSL files are located in `/etc/pki/tls`.
+
+NOTE: The above output is from Fedora 24, it should be similar on other operative systems.
+
+WARNING: It seems that if you are using RVM to manage your Ruby versions, you should ignore this
+and use: $rvm_path/usr/ssl
+
+== Hand over your certificate to OpenSSL in PEM format and rehash.
+`c_rehash` specs the certificates in PEM format, so before proceeding we need make sure is in PEM format. If you used the guide
```

Comment:
This applies in general, the posts uses `we` and `you` in the same paragraph. I recommend avoid this. For example, you can keep `we` and transform the `you` to third person speech (with `it`).

For example: 
`If you used the guide for securing communications on Hawkular, you already have the certificates in DER ...` 

`The guide for securing communications on Hawkular produces the certificates in DER ...`

Also, link as much as possible, in this case link to the guide from here.